codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/CborParserGenerator.kt

```diff
@@ -206,33 +206,59 @@
+                                    val symbol = symbolProvider.toSymbol(member)
+                                    if (symbol.isRustBoxed()) {
+                                        rustBlock("") {
+                                            rustTemplate("let v = #{DeserializeMember:W}?;",
+                                                "DeserializeMember" to deserializeMember(member))
+
+                                            for (customization in customizations) {
+                                                customization.section(
+                                                    CborParserSection.BeforeBoxingDeserializedMember(
+                                                        member
+                                                    )
+                                                )(this)
+                                            }
+                                            rust("Box::new(v)")
                                         }
-                                        for (customization in customizations) {
-                                            customization.section(CborParserSection.BeforeBoxingDeserializedMember(member))(this)
-                                        }
-                                        rust("Box::new(v)")
+                                    } else {
+                                        rustTemplate("#{DeserializeMember:W}?",
+                                            "DeserializeMember" to deserializeMember(member))
                                     }
-                                } else {
-                                    deserializeMember(member)
-                                    rust("?")
                                 }
                             }
                         }
+
+                        if (member.isOptional) {
+                            // Call `builder.set_member()` only if the value for the field on the wire is not null.
+                            rustTemplate(
+                                """
+                                ::aws_smithy_cbor::decode::set_optional(builder, decoder, |builder, decoder| {
+                                    Ok(#{MemberSettingWritable:W})
+                                })?
+                                """,
+                                "MemberSettingWritable" to callBuilderSetMemberFieldWritable
+                            )
+                        }
+                        else {
+                            callBuilderSetMemberFieldWritable.invoke(this)
+                        }
                     }
                 }
 
-                // TODO Skip like in JSON or reject? I think we should reject in JSON too. Cut issue.
-                rust("_ => { todo!() }")
+                rust(
+                    """
+                    _ => { 
+                        decoder.skip()?;
+                        builder
+                    }
+                    """)
```

Comment:
Let's add a comment that we're explicitly skipping unrelated fields as per the spec.

I guess our current behavior in JSON protocols of skipping too is desirable then. I opened https://github.com/smithy-lang/smithy-rs/issues/2598 in the past but never followed up on it. It would be good to justify why we're skipping and link to the spec.

style nit:

```suggestion
                rust(
                    """
                    _ => { 
                        decoder.skip()?;
                        builder
                    }
                    """
                )
```

Doesn't ktlint catch these?